executor/brie.go

```diff
@@ -408,6 +703,47 @@ func (gs *tidbGlueSession) Execute(ctx context.Context, sql string) error {
 	return err
 }
 
+func (gs *tidbGlueSession) ExecuteWithLog(ctx context.Context, sql string, purpose string, logger importlog.Logger) error {
+	se, err := gs.GetSession()
+	if err != nil {
+		return err
+	}
+	defer se.Close()
+	return common.Retry(purpose, logger, func() error {
+		_, err := se.Execute(ctx, sql)
```

Comment:
`se.Execute` will not execute the SQL actually if the sql is `SELECT ... FROM`, can we also use `RestrictedExec` here?

Reply:
for comments till this one, https://github.com/pingcap/tidb/pull/21195/commits/760942c949e4a3bade9c4764e920a794eb483551 should have made enough modifications.